lib/membrane_live/event.ex

```diff
@@ -0,0 +1,239 @@
+          cert: state.network_options[:dtls_cert]
+        ]
+      else
+        [
+          client_mode: false,
+          dtls_srtp: true
+        ]
+      end
+
+    endpoint = %WebRTC{
+      rtc_engine: rtc_engine,
+      ice_name: peer.id,
+      owner: self(),
+      integrated_turn_options: state.network_options[:integrated_turn_options],
+      integrated_turn_domain: state.network_options[:integrated_turn_domain],
+      handshake_opts: handshake_opts,
+      log_metadata: [peer_id: peer.id],
+      trace_context: state.trace_ctx,
+      webrtc_extensions: [Mid, TWCC],
+      peer_metadata: peer.metadata,
+      filter_codecs: fn {rtp, fmtp} ->
+        case rtp.encoding do
+          "opus" -> true
+          "H264" -> fmtp.profile_level_id === 0x42E01F
+          _unsupported_codec -> false
+        end
+      end
+    }
+
+    Engine.accept_peer(rtc_engine, peer.id)
+    :ok = Engine.add_endpoint(rtc_engine, endpoint, peer_id: peer.id, node: peer_node)
+
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_info(%Message.PeerLeft{peer: peer}, state) do
+    Membrane.Logger.info("Peer #{inspect(peer.id)} left RTC Engine")
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_info(%Message.EndpointCrashed{endpoint_id: endpoint_id}, state) do
+    Membrane.Logger.error("Endpoint #{inspect(endpoint_id)} has crashed!")
+    peer_channel = state.peer_channels[endpoint_id]
+
+    error_message = "WebRTC endpoint has crashed, please refresh the page to reconnect"
+    data = MediaEvent.create_error_event(error_message)
+    send(peer_channel, {:media_event, data})
+
+    {:noreply, state}
+  end
+
+  # media_event coming from client
+  @impl true
+  def handle_info({:media_event, _from, _event} = msg, state) do
+    Engine.receive_media_event(state.rtc_engine, msg)
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_info({:DOWN, _ref, :process, pid, _reason}, state) do
+    if pid == state.rtc_engine do
+      event_span_id(state.event_id)
+      |> Membrane.OpenTelemetry.end_span()
+
+      {:stop, :normal, state}
+    else
+      {peer_id, _peer_channel_id} =
+        state.peer_channels
+        |> Enum.find(fn {_peer_id, peer_channel_pid} -> peer_channel_pid == pid end)
+
+      Engine.remove_peer(state.rtc_engine, peer_id)
+      {_elem, state} = pop_in(state, [:peer_channels, peer_id])
+
+      if state.peer_channels == %{}, do: Engine.terminate(state.rtc_engine)
+
+      {:noreply, state}
+    end
+  end
+
+  defp tracing_metadata(),
+    do: [
+      {:"library.language", :erlang},
+      {:"library.name", :membrane_rtc_engine},
+      {:"library.version", "server:#{Application.spec(:membrane_rtc_engine, :vsn)}"}
+    ]
+
+  defp event_span_id(id), do: "event:#{id}"
+
+  @impl true
+  def handle_info({:playlist_playable, :audio, _playlist_idl}, state) do
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_info({:playlist_playable, :video, playlist_idl}, state) do
+    {:noreply, state}
+  end
```

Comment:
I would add some comment that this will be used for detecting when the HLS stream started.